test/WebJobs.Script.Tests/Description/DotNet/RawAssemblyCompilationTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Description.DotNet
+{
+    public class RawAssemblyCompilationTests
+    {
+        [Theory]
+        [MemberData(nameof(GetTargetNames))]
+        public void GetEntryPointSignature_BindsToExpectedMethod(string entryPointName, string methodName)
+        {
+            var testAssembly = typeof(TestFunction1).Assembly;
+            string assemblyPath = new Uri(testAssembly.CodeBase, UriKind.Absolute).LocalPath;
+            var compilation = new RawAssemblyCompilation(assemblyPath, entryPointName);
+
+            FunctionSignature signature = compilation.GetEntryPointSignature(new FunctionEntryPointResolver(), testAssembly);
+
+            Assert.NotNull(signature);
+            Assert.Equal(methodName, signature.MethodName);
+        }
+
+        [Fact]
+        public void GetEntryPointSignature_PrefersStaticMethod()
+        {
+            var testAssembly = typeof(TestFunction1).Assembly;
+            string assemblyPath = new Uri(testAssembly.CodeBase, UriKind.Absolute).LocalPath;
+            var compilation = new RawAssemblyCompilation(assemblyPath, $"{typeof(TestFunction3).FullName}.{nameof(TestFunction3.Run)}");
+
+            FunctionSignature signature = compilation.GetEntryPointSignature(new FunctionEntryPointResolver(), testAssembly);
+
+            Assert.NotNull(signature);
+            Assert.Equal(nameof(TestFunction3.Run), signature.MethodName);
```

Comment:
Is this verifying that the static method was resolved? It looks like it's just checking the name "Run".

Reply:
If someone swapped the resolution ordering (for some reason), this would still pass as resolution would still work. I would expect the test to fail as we expect the static method to resolve first. I'm not sure that'll ever happen, but thought I'd call it out :-)

Reply:
Yes, the additional check will prevent that. 

That would be a very odd change to make without expecting behavior to change in a breaking way, but I agree a test should fail if that happens :)